### PR TITLE
Add hash as CLI function

### PR DESCRIPTION
**Pull request by gavofyork**
A bit of an odd function, but:
- there are few CLI sha3 utils;
- even fewer cross-platform;
- none (as far as I can tell) that do original Keccak (like us) rather than the official SHA3 variant;
- such hashes are needed for using our URL hinting infrastructure.

### CODE REVIEW

**Comment by rphmeier** [comment]
This should be in the `ethkey` binary, not parity proper.

**Comment by tomusdrw** [comment]
`rhash --sha3-256` works fine, perhaps registering github dapp would be more suitable for `parity` binary?

`parity dapp-register ethcore/daoclaim <commithash>`

File: parity/main.rs [hunk 1]
```diff
@@ -85,17 +85,32 @@ mod sync;
...
 use cli::print_version;
 use configuration::{Cmd, Configuration};
 use deprecated::find_deprecated;
 
+fn print_hash_of(maybe_file: Option<String>) -> Result<String, String> {
+	if let Some(file) = maybe_file {
+		let mut f = try!(File::open(&file).map_err(|_| "Unable to open file".to_owned()));
+		let mut bytes = vec![];
+		try!(f.read_to_end(&mut bytes).map_err(|_| "Unable to read from file".to_owned()));
+		Ok(bytes.sha3().hex())
```
Comment (debris) [inline parity/main.rs]: I think it's sensible to use the function created by @tomusdrw in other pr (once it's merged) https://github.com/ethcore/parity/pull/1993/files#diff-ce115bfe675e73b03323a5a2d6d2ab84R70

Comment (gavofyork) [inline parity/main.rs]: indeed

**Comment by gavofyork** [comment]
yeah i don't think `ethkey` is much more suited - that's meant more for key management

**Comment by rphmeier** [comment]
I like @tomusdrw's suggestion of `parity tools`

**Comment by gavofyork** [comment]
appveyor appears spurious.

**Pull request merged**